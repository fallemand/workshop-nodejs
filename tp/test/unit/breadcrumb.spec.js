const React = require('react');
const renderer = require('react-test-renderer');
const Breadcrumb = require('../../views/components/Breadcrumb');

describe('Breadcrumb (Snapshot)', () => {

    const mockedCategory = [ 
        { id: 'MLA3937', name: 'Joyas y Relojes' },
        { id: 'MLA1431', name: 'Joyas' },
        { id: 'MLA1438', name: 'Anillos' },
        { id: 'MLA23107', name: 'Acero' },
        { id: 'MLA26164', name: 'Otras Piedras' } 
    ];
  
    test('Breadcrumb renders without results', () => {
        const component = renderer.create(<Breadcrumb category="" query="anil"/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
      });

      test('Breadcrumb renders with results', () => {
        const component = renderer.create(<Breadcrumb category={mockedCategory} query="anil"/>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
      });
  
  });