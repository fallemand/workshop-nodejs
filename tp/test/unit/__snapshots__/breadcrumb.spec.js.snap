// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb (Snapshot) Breadcrumb renders item 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__element"
  >
    <a
      href="/app"
    >
      Inicio
    </a>
  </li>
  <li
    className="breadcrumb__element"
  >
    Celulares y Telefonía
  </li>
  <li
    className="breadcrumb__element"
  >
    Celulares
  </li>
  <li
    className="breadcrumb__element"
  >
    Iphone
  </li>
</ul>
`;

exports[`Breadcrumb (Snapshot) Breadcrumb renders search 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__element"
  >
    <a
      href="/app"
    >
      Inicio
    </a>
  </li>
  <li
    className="breadcrumb__element"
  >
    Celulares y Telefonía
  </li>
  <li
    className="breadcrumb__element"
  >
    Celulares
  </li>
  <li
    className="breadcrumb__element"
  >
    Iphone
  </li>
  <li
    className="breadcrumb__element"
  >
    "
    iphone 7
    "
  </li>
</ul>
`;
